added checklists about builder and files (#6)
mate-academy/jv-program-common-mistakes: java-core/builder-file/builder.md

### Common mistakes

#### Using private constructor of class 

> If you use builder pattern better make constructor of outer class with a Builder as an input parameter private.

Reviewer: ```suggestion
If you use a builder pattern, make the outer class's constructor with a Builder input parameter private, not public, or default.
```
